### PR TITLE
Use posix functions instead of macros.

### PR DESCRIPTION
**Pull request by nmoinvaz**

### CODE REVIEW

File: prioritytask.c [hunk 1]
```diff
@@ -25,14 +25,16 @@ static bool PriorityTask_Notification_OnTaskStart(void *UserPtr, const char *Typ
     char Level[320] = {0};
 
     if (ITask_GetStringArgument(PriorityTask->TaskHandle, "level", Level, sizeof(Level)) == false)
-        String_CopyLength(Level, "low", sizeof(Level));
+        strncpy(Level, "low", sizeof(Level));
```
Comment (sergio-nsk) [inline prioritytask.c:28]: Why not strict `strncpy_s`?

Comment (nmoinvaz) [reply prioritytask.c]: Fixed.

**Comment by sergio-nsk** [comment]
macOS is C99?
> error: implicit declaration of function 'strcpy_s' is invalid in C99

**Comment by nmoinvaz** [comment]
Even using `C_STANDARD 11` it still does not work weird.

**Comment by sergio-nsk** [comment]
> As with all bounds-checked functions, `strcpy_s` is only guaranteed to be available if `__STDC_LIB_EXT1__` is defined by the implementation and if the user defines `__STDC_WANT_LIB_EXT1__` to the integer constant 1 before including `<string.h>`.

If it does not help :-(

**Comment by sergio-nsk** [comment]
Maybe https://stackoverflow.com/questions/16700541/is-support-of-annex-k-in-c11-required-for-a-conforming-implementation
> An implementation that defines `__STDC_LIB_EXT1__` must define it as `201112L`; both N1570 and the released C11 standard got this wrong, but it was fixed in a Technical Corrigendum.

**Comment by nmoinvaz** [comment]
I have just reverted back to defining `_CRT_SECURE_NO_WARNINGS` for Windows, because macOS does not support `strcpy_s`.

**Review by sergio-nsk — APPROVED** [review]

**Pull request merged**